starboard/socket.h

```diff
@@ -126,6 +137,41 @@ typedef struct SbSocketResolution {
 // Well-defined value for an invalid socket handle.
 #define kSbSocketInvalid ((SbSocket)NULL)
 
+#if SB_API_VERSION >= 16
```

Comment:
This code should go in starboard/common. 
We don't allow c++ code in the starboard/foo.h headers.